Ginger/DatabaseUnitTest/DataBaseUnitTest.cs

```diff
@@ -0,0 +1,151 @@
+*/
+#endregion
+
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MSAccessDB;
+using System.Collections.Generic;
+using System;
+using GingerTestHelper;
+
+namespace GingerCoreNETUnitTest.Database
+{
+
+    [TestClass]
+
+    public class DataBaseUnitTest
+    {
+        public static MSAccessDBCon db = new MSAccessDBCon();
+
+        [ClassInitialize]
+        public static void ClassInit(TestContext context)
+        {
+            Dictionary<string, string> param = new Dictionary<string, string>();
+            string FilePath = TestResources.GetTestResourcesFile(@"SignUp.accdb");
+            param.Add("ConnectionString", @"Provider=Microsoft.ACE.OLEDB.12.0;Data Source=" + FilePath +";");
+            db.KeyvalParamatersList = param;
+            Boolean testconn = db.OpenConnection(param);
+            
+        }
+
+        [TestMethod]
+        public void OpenConnection()
+        {
+            //Arrange
+            Dictionary<string, string> param = new Dictionary<string, string>();
+            param.Add("ConnectionString", @"Provider=Microsoft.ACE.OLEDB.12.0;Data Source= C:\GingerSourceControl\Solutions\Ginger_Regression_Testing\Documents\WOEI\SignUp.accdb;");
+
+            //Act
+            Boolean testconn = db.OpenConnection(param);
+
+            //Assert
+            Assert.AreEqual(testconn, true);
+        }
+
+        [TestMethod]
+        public void GetTableList()
+        {
+            //Arrange
+            
+            List<string> Tables= null;
+            
+            //Act
+
+             Tables= db.GetTablesList();
+           
+            //Assert
+            Assert.AreEqual(Tables.Count, 2);
+        }
+
+        [TestMethod]
+        public void GetTablesColumns()
+        {
+            //Arrange
+            List<string> Columns = null;
+            string tablename = "Person";
+            
+            //Act
+            Columns = db.GetTablesColumns(tablename);
+           
+            //Assert
+            Assert.AreEqual(Columns.Count, 9);
+        }
+
+        [TestMethod]
+        public void RunUpdateCommand()
+        {
+            //Arrange
+            string upadateCommand = "update Person set LName=\"EFGH\" where ID=1";
+            string result = null;
+            
+            //Act
+            result = db.RunUpdateCommand(upadateCommand, false);
+           
+            //Assert
+            Assert.AreEqual(result, "1");
+        }
+
+        [TestMethod]
+        public void GetSingleValue()
+        {
+            //Arrange
+            string result = null;
+            
+            //Act
+            result = db.GetSingleValue("Person", "FName", "ID=2");
+            
+            //Assert
+            Assert.AreEqual(result, "LMPO");
+        }
+
+        [TestMethod]
+        public void DBQuery()
+        {
+            //Arrange
+            List<object> result = null;
+            
+            //Act
+            result = db.DBQuery("Select * from Person");
```

Comment:
DBQuery needs to return a table no List<object> - let's discuss

Reply:
please handle

Reply:
handled